src/main/scala/io/github/sentenza/hacktoberfest18/sort/CommonSortings.scala

```diff
@@ -25,7 +25,18 @@ object CommonSortings {
   }
 
   object CommonSorting {
-    def bubbleSort(array: Array[Int]): Array[Int] = ???
+    def bubbleSort(array: Array[Int]): Array[Int] = {
+
+        def go(a: Array[Int], i: Int): Array[Int] = {
+            if(i+1 >= a.length) a
+            else if(a(i) > a(i+1)) go(a.patch(i, Array(a(i+1), a(i)), 2), 0)
```

Comment:
Can you think about an alternative version that doesn't involve the mutation of the array (not using [Array.patch()][patch])?

[patch]: https://www.scala-lang.org/api/2.12.1/scala/Array.html#patch(from:Int,that:scala.collection.GenSeq[A],replaced:Int):Array[A]

Reply:
If i understand the documentation correctly, Array.patch returns a new Array without mutating the source Array. [Array.patch](https://www.scala-lang.org/api/2.12.6/scala/Array.html#patch(from:Int,that:scala.collection.GenSeq[A],replaced:Int):Array[A])

Reply:
You're absolutely right. My bad, nevermind about that.